lib/gs-plugin-loader.c

```diff
@@ -2527,13 +2505,11 @@ mount_removed_cb (GVolumeMonitor *volume_monitor,
 		  GsPluginLoader *plugin_loader)
 {
 	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
-	GMount *existing_mount = find_mount_in_list (priv->removable_mounts,
-						     mount);
-	if (existing_mount) {
-		priv->removable_mounts = g_list_remove (priv->removable_mounts,
-							existing_mount);
+	GList *list_orig = priv->removable_mounts;
+
+	priv->removable_mounts = g_list_remove (priv->removable_mounts, mount);
+	if (list_orig != priv->removable_mounts)
```

Comment:
This means the notify will only happen if the mount that got removed was the first of the list, is this the intention? I think it should always notify if one of the mount points got removed, right? (it's also what the code was doing before)

Reply:
Ah, I mixed that up slightly. Yes, it should happen as long as there was any change, so I'm now comparing based on length before and after the removal. Fixed locally.